### DISCUSSION TITLE
{name} in alert message

### DISCUSSION THREAD

**Discussion by Spinestars** [apache/skywalking#10606] [Q&A] [answered]
endpoint_abnormal_rule:
            metrics-name: endpoint_abnormal
            op: ">="
            threshold: 5
            period: 1
            count: 3
            silence-period: 5
            message: scope type：endpoint<br> <font color=#FF0000> {name} </font> http code error.
---
In the above example, where does {name} in message come from?
Suppose I wanted to change "scope type: endpoint" to "scope type: {type}. What should I do?

**Comment by wu-sheng** [comment, marked as answer]
You don't need that. Alerting is metrics oriented, and the metric literal name represents the scope. There is no point to provide a variable of `{type}`.

**Reply by Spinestars** [reply]
To be precise, {type} is not what I really need. I want to output the detailed alarm value in message.



Just like in prometheus, I can use "$value" to get the current value of the alarm indicator, which gives me a better idea of the service status.

**Reply by wu-sheng** [reply]
There isn't value. The threshold is at a range. [T1 ... T5] have [Value1 ... Value5]. You defined the `count` to indicate how many times you expect to match.

Prometheus is Promtheus. Different systems have different ways to work.

**Reply by Spinestars** [reply]
Okay. Thank you

**Reply by Jack-Ye** [reply]
> There isn't value. The threshold is at a range. [T1 ... T5] have [Value1 ... Value5]. You defined the `count` to indicate how many times you expect to match.
> 
> Prometheus is Promtheus. Different systems have different ways to work.

why not return the `matchCount` in `window` to alarmMessage, since the data is there, just try to pass the data.
